python/cudf/cudf/tests/test_replace.py

```diff
@@ -236,11 +250,26 @@ def test_dataframe_replace(df, to_replace, value):
     else:
         gd_to_replace = to_replace
 
-    if pd_value is None:
-        expected = pdf.replace(to_replace=pd_to_replace)
-    else:
-        expected = pdf.replace(to_replace=pd_to_replace, value=pd_value)
-    actual = gdf.replace(to_replace=gd_to_replace, value=gd_value)
+    with expect_warning_if(
+        PANDAS_GE_220
+        and isinstance(df["a"].dtype, cudf.CategoricalDtype)
+        and isinstance(to_replace, str)
```

Comment:
This check is redundant with the equality comparison below.

Reply:
There are similar checks in other test cases too.

Reply:
This was mean to protect against comparing a sting to a `Series` which results in a `Series`, and `bool(Series)` here would raise an error here. (same as below)